components/datepicker/DatePicker.js

```diff
@@ -0,0 +1,65 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import DayPicker from 'react-day-picker';
+import NavigationBar from './NavigationBar';
+import WeekDay from './WeekDay';
+import { convertModifiersToClassnames } from './utils';
+import cx from 'classnames';
+import theme from './theme.css';
+
+class DatePicker extends PureComponent {
+  state = {
+    selectedDate: null,
+  };
+
+  componentWillMount() {
+    const { selectedDate } = this.props;
+
+    if (selectedDate) {
+      this.setState({ selectedDate });
+    }
+  }
+
+  handleDayClick = day => {
+    this.setState(
+      {
+        selectedDate: day,
+      },
+      this.props.onChange({ selectedDate: day }),
```

Comment:
this is defined as a callback to setState but won't it be executed immediately as it's not wrapped in a function?

Reply:
I solved this by using the `componentDidUpdate` method as @duivvv mentioned.
```
componentDidUpdate() {
    this.props.onChange({ selectedDate: this.state.selectedDate });
  }
```